testslide/matchers.py

```diff
@@ -53,6 +53,8 @@ class _AndMatcher(_AlreadyChainedMatcher):
     """
 
     def __init__(self, a: Matcher, b: Matcher) -> None:
+        if type(a) != type(Matcher()) or type(b) != type(Matcher()):
```

Comment:
I'm wondering if we should be testing types via `isinstance` instead of exact type as in here

```suggestion
        if not isinstance(a, Matcher) or not isinstance(b, Matcher)):
```

The same would apply to the other cases as well

NOTE: I haven't yet checked the implications of the changes.

Reply:
Thanks for your reply, I'll get it done. Right now now I am facing some issues setting up the development environment. The moment I get it done. I'll implement the suggested changes. 😄

Reply:
Done! 😄
@macisamuele 

I added a [new PR](https://github.com/facebook/TestSlide/pull/290) after resolving the installation and environment issues.

Reply:
We're working on making it better and streamlined.
I would suggest to create a virtual environment, activate it and the run `make install_build_deps install_deps` so all the dependencies will be available there

Reply:
I did the same. But it threw an error stating that those modules were not found. I had to explicitly install `typeguard` and `psutil` to make it work.